bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/WorkflowDelegationStep.java

```diff
@@ -78,125 +103,217 @@
+      WorkflowModel workflowModel = null;
+      final String debugText = "Using configured delegate Workflow Model [ {} ] found in {}";
+      final String exceptionText = "Could not find configured Workflow Model [ %s ] in %s";
+
+      // Check /conf
+      if (StringUtils.isNotBlank(confWorkflowModelId)) {
+        workflowModel = getWorkflowModel(workflowSession, confWorkflowModelId);
+
+        if (workflowModel != null) {
+          log.debug(debugText, workflowModel.getId(), "/conf");
+        } else {
+          throw new WorkflowException(String.format(exceptionText, confWorkflowModelId, "/conf"));
         }
+        // Check Content
+      } else if (StringUtils.isNotBlank(contentWorkflowModelId)) {
 
-        log.debug("Provided PROCESS_ARGS: propertyName = [ {} ], Default Workflow Model = [ {} ]", propertyName, defaultWorkflowModelId);
-
-        ResourceResolver resolver = null;
-        WorkflowData wfData = workItem.getWorkflowData();
+        workflowModel = getWorkflowModel(workflowSession, contentWorkflowModelId);
 
-        if (!workflowHelper.isPathTypedPayload(wfData)) {
-            log.warn("Could not locate a JCR_PATH payload for this workflow. Skipping delegation.");
-            return;
+        if (workflowModel != null) {
+          log.debug(debugText, workflowModel.getId(), "/content");
+        } else {
+          throw new WorkflowException(String.format(exceptionText, confWorkflowModelId, "/content"));
         }
 
-        final String path = wfData.getPayload().toString();
+        // Check default
+      } else if (StringUtils.isNotBlank(defaultWorkflowModelId)) {
+        workflowModel = getWorkflowModel(workflowSession, defaultWorkflowModelId);
 
-        // Get the resource resolver
-
-        resolver = workflowHelper.getResourceResolver(workflowSession);
-        if (resolver == null) {
-            throw new WorkflowException("Could not adapt the WorkflowSession to a ResourceResolver. Something has gone terribly wrong!");
+        if (workflowModel != null) {
+          log.debug(debugText, workflowModel.getId(), "parameter defaultworkflowmodelid");
+        } else {
+          throw new WorkflowException(
+              String.format(exceptionText, confWorkflowModelId, "parameter defaultworkflowmodelid"));
         }
+      }
 
-        // Derive the Page or Asset resource so we have a normalized entry-point for finding the /jcr:content resource
+      return workflowModel;
+    }
 
-        final Resource pageOrAssetResource = workflowHelper.getPageOrAssetResource(resolver, path);
-        if (pageOrAssetResource == null) {
-            log.warn("Could not resolve [ {} ] to an Asset or Page. Skipping delegation.", path);
-            return;
+    /**
+     * Ensures that workflow model id points to $model$/jcr:content/model
+     */
+    private WorkflowModel getWorkflowModel(WorkflowSession workflowSession, String workflowModelId) {
```

Comment:
not 100% necessary, but this seems like a good method to have in https://adobe-consulting-services.github.io/acs-aem-commons/apidocs/com/adobe/acs/commons/util/WorkflowHelper.html

Swear ive written ~ the same code before in this code-base and its a pretty common use-case.